lib/internal/worker.js

```diff
@@ -314,6 +316,16 @@ class Worker extends EventEmitter {
 
     return makeResourceLimits(this[kHandle].getResourceLimits());
   }
+
+  getHeapSnapshot() {
+    const heapSnapshotTaker = this[kHandle] && this[kHandle].takeHeapSnapshot();
+    if (!heapSnapshotTaker) return Promise.reject(new ERR_WORKER_NOT_RUNNING());
```

Comment:
you could move this line into the promise and call reject

Reply:
Sure, done, although it really shouldn’t make any difference ;)